### ISSUE TITLE
The disk import via 'DataVolume' fails for a warm migration of windows VM.

### ISSUE THREAD

**Issue by cniket** [kubevirt/containerized-data-importer#3430]
**What happened**:
I have a DV that has source as `vddk` (`spec.source.vddk`) using which I am trying to migration a windows VM from VMware to Kubevirt. Before starting the migration if the windows VM is in 'power on' state, then the migration(warm) doesn't start and the corresponding import pod fails. Attaching the file: [importer-plan-winserver2k19-warm-powered-on.log](https://github.com/user-attachments/files/16945121/importer-plan-winserver2k19-warm-powered-on.log)  that has the corresponding errors.
If the windows VM is in 'power-off' state before starting the migration, then the migration(cold) happens without any issue.
For Linux VM, both 'warm' and 'cold' migration works without any issue.

Tried the migration with VSphere administration privileges(full access), however still getting the same errors.

**What you expected to happen**:
Both 'cold' and 'warm' migration for Windows VM should work similar to Linux VMs.

**How to reproduce it (as minimally and precisely as possible)**:
- `DV` with `spec.source.vddk` is getting autocreated similar to this: https://github.com/kubevirt/containerized-data-importer/blob/main/doc/datavolumes.md#vddk-data-volume  after creating a [forklift](https://github.com/kubev2v/forklift/blob/main/operator/docs/k8s.md) `Migration` CR.  The resulted DV's disk import fails for windows VM if the VM is in power-on state,  however the same windows VM's cold migration works and for the Linux VMs both 'cold' and 'warm' works as explained earlier.

**Additional context**:

**Environment**:
- CDI version (use `kubectl get deployments cdi-deployment -o yaml`):  

```
❯ k get cdi cdi -oyaml
apiVersion: cdi.kubevirt.io/v1beta1
kind: CDI
metadata:
  annotations:
    cdi.kubevirt.io/configAuthority: ""
  creationTimestamp: "2024-05-03T17:53:01Z"
  finalizers:
  - operator.cdi.kubevirt.io
  generation: 29
  name: cdi
  resourceVersion: "123180875"
  uid: adca7d62-bb11-4a86-ac47-93017635d0d9
spec:
  imagePullPolicy: IfNotPresent
  infra:
    nodeSelector:
      kubernetes.io/os: linux
    tolerations:
    - key: CriticalAddonsOnly
      operator: Exists
  workload:
    nodeSelector:
      kubernetes.io/os: linux
status:
  conditions:
[...]
    type: Degraded
  observedVersion: v1.57.0
  operatorVersion: v1.57.0
  phase: Deployed
  targetVersion: v1.57.0
❯

```

- Kubernetes version (use `kubectl version`): 

```
❯ kubectl version
Client Version: v1.30.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.29.2
❯
```

- DV specification: 

```
❯ k get dv plan-winserver2k19-vm-3187-fmdg2 -n virtualmachines -oyaml
apiVersion: cdi.kubevirt.io/v1beta1
kind: DataVolume
metadata:
  annotations:
    cdi.kubevirt.io/storage.bind.immediate.requested: "true"
    cdi.kubevirt.io/storage.deleteAfterCompletion: "false"
    cdi.kubevirt.io/storage.usePopulator: "false"
    forklift.konveyor.io/disk-source: '[vcenter-datastore] winserver2k19/winserver2k19_1.vmdk'
    migration: 325eb7fc-e733-4a25-b105-db3422813752
    plan: 8ff9313b-4f38-4104-b623-9cc63fb17db6
    vmID: vm-3187
  generateName: plan-winserver2k19-vm-3187-
  generation: 1
  labels:
[...]
spec:
  checkpoints:
  - current: snapshot-5368
    previous: ""
  source:
    vddk:
      backingFile: '[vcenter-datastore-1] winserver2k19/winserver2k19_1.vmdk'
      initImageURL: jmv2/vddk:7.0.3
      secretRef: plan-winserver2k19-vm-3187-b5df3
      thumbprint: <thumbprint>
      url: <vsphere-url>/sdk
      uuid: <vm-uuid>
  storage:
    accessModes:
    - ReadWriteMany
    resources:
      requests:
        storage: 1Gi
    storageClassName: ceph-block
status:
[...]
❯
```

- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

**Comment by mrnold** [comment]
This looks very similar to a bug that was recently fixed: https://github.com/kubevirt/containerized-data-importer/pull/3385

Is there any chance you can try with a version that includes that change? It looks like it was backported to CDI releases 1.58, 1.59, and 1.60.

**Comment by cniket** [comment]
Hello @mrnold ,

Thanks a lot for your reply and pointing to the respective PR.

I will try with one of those CDI releases with higher version and will share the results.

**Comment by cniket** [comment]
Hello @mrnold ,

The migration completed successfully without those errors with cdi v1.58. Thanks a lot for your help.

However, I am facing new issue now, after completing DV migration, the VM comes up in 'Running' state. When I am trying to take its console using `kubectl virt vnc <vm-name>` , getting windows blue screen error as per attached screenshot. (Not sure if this is the right place to add this issue)

<img width="1279" alt="Screenshot 2024-09-12 at 12 42 02" src="https://github.com/user-attachments/assets/1d611a99-0bfd-4ac2-ba4b-6beb8637f70c">


JFI: serureBoot is kept 'false' in the VM template;
```
        firmware:
          bootloader:
            efi:
              secureBoot: false
```
What could be cause that boot issue for post migration?

**Comment by akalenyu** [comment]
long shot but maybe @lyarwood or @vsibirsk may be able to help

**Comment by mrnold** [comment]
Usually that means something didn't work right in the VM conversion step, either the VirtIO drivers weren't installed or the virtual hardware configuration was not quite as expected. Probably forklift is the right place to continue discussion.

**Comment by cloudcafetech** [comment]
@cniket Any idea how to VMware VM (Linux) UUID?

Getting following error ... 

```
Defaulted container "importer" out of: importer, vddk-side-car (init)
I0212 11:59:42.603776       1 importer.go:107] Starting importer
I0212 11:59:42.604760       1 importer.go:182] begin import process
I0212 11:59:42.604795       1 vddk-datasource_amd64.go:831] Creating VDDK data source: backingFile [[datastore1] vw-jumphost/vw-jumphost.vmdk], currentCheckpoint [], previousCheckpoint [], finalCheckpoint []
I0212 11:59:42.713152       1 vddk-datasource_amd64.go:488] VM 1c724d56-4337-7af1-7a83-651fade176fd not found in datacenter Datacenter:datacenter-3 @ /pkdc.
E0212 11:59:42.713207       1 vddk-datasource_amd64.go:288] Unable to find MORef for VM with UUID 1c724d56-4337-7af1-7a83-651fade176fd!
E0212 11:59:42.713226       1 vddk-datasource_amd64.go:841] Unable to log in to VMware: unable to locate VM in any datacenter
E0212 11:59:42.713241       1 importer.go:347] unable to locate VM in any datacenter
```

**Comment by mrnold** [comment]
@cloudcafetech I saw something like that awhile back with nested datacenters. Are you able to share your datacenter layout? Maybe there's something there that CDI doesn't expect.

**Comment by cloudcafetech** [comment]
@mrnold I am using in Homelab.

One ESXI (v7) server where source VM running.

K8s Setup:
One Proxmox server where one k8s Master running using Talos 1.92
Two worker node (baremetal) running using Talos 1.92

K8s: Longhorn, NFS-CSI, NFS provisioner.

Trying to migrate vm from ESXI to kubevirt by importing vmdk.

**Comment by mrnold** [comment]
That doesn't sound like anything unusual, how are you getting the VM's UUID?
